README.md
# TierraLuna

Este proyecto comenzó como una serie de tutoriales creados por el Equipo de Huayra GNU/Linux para aprender programación de videojuegos utilizando **PILAS-engine** y **Huayra**. 
Se trata de un viaje a la luna donde se pondrá a prueba nuestra habilidad, paciencia y temple. **TierraLuna** es un juego para leer, ya que en el camino aprenderemos conceptos de astronomía, geografía, física y ecología, mientras intentamos superar los cinco niveles y alunizar.

![](imagenes/preview.png)
![](imagenes/preview3.png)

## Autores

**TierraLuna** fue realizado por

* Diego Accorinti
* Claudio *Maléfico* Andaur
* Rodrigo Manresa

para Huayra GNU/Linux. 

## Temas

En **Tierra-Luna** podemos elegir *temas* que modifican los gráficos del juego con distintas estéticas. Tanto los sprites (naves y objetos móviles) como los decorados, pueden modificarse eligiendo el tema que más te guste. 
Pero no sólo los gráficos pueden cambiarse, también puede elegirse el texto que acompaña al jugador durante el viaje. De esta manera, Tierra-Luna puede utilizarse para contar practicamente cualquier cosa, jugando. 
Se distribuyen dos temas con Tierra-Luna, el tema original que nos cuenta sobre astronomía a medida que viajamos hacia la Luna, y el tema "Steampunk" que nos cuenta sobre la literatura de ciencia-ficción del S.XIX.

![](imagenes/preview4.jpg)
![](imagenes/preview5.jpg)

## Cómo escribir tus propias aventuras con **Tierra-Luna**

Copia cualquiera de las carpetas dentro del directorio *`temas`*, y renómbralo.
Puedes modificar o reemplazar varios o todos los gráficos de la carpeta `imagenes` (mantén los nombres de los archivos).
Escribe el texto de tu aventura como texto plano usando un editor de texto. Puedes usar las siguientes etiquetas al comienzo de una linea para indicar situaciones especiales:

Utiliza la etiqueta `<NIVEL#>` para indicar el tipo de nivel que deseas cargar (valores aceptados son `<NIVEL1>`, `<NIVEL2>` hasta `<NIVEL5>`).
Utiliza la etiqueta `<EVENTO#>` para que aparezca o bien un "satelite" o bien una "estacion reparadora" (`<EVENTO1>` y `<EVENTO2>`)
Utiliza la etiqueta `<FIN>` para indicar cuando el juego debe terminar su recorrido.
Utiliza la etiqueta `<LOOP>` para indicar que todo debe volver a empezar.

Por ejemplo la linea:

```
<NIVEL2>Un nivel cualquiera</NIVEL2>
```

Haría que se imprima el texto "Un nivel cualquiera", al mismo tiempo que cambiarian los fondos y sprites del juego por los que corresponden a una pantalla de nivel 2.

## Instalación del tema

Para que **Tierra-Luna** pueda reconocer correctamente tu tema, debes editar el archivo *temas-dic.py*, indicando el nombre del tema, y los items que incluye. 
Puedes modificar unicamente los textos por ejemplo y utilizar los graficos de otros temas.

## Licencia

Software libre bajo licencia GPL3 

This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.


debian/changelog
tierraluna (0.2.8) unstable; urgency=medium

  * fix lintian 

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 08 Nov 2018 14:54:02 -0300

tierraluna (0.2.7) unstable; urgency=medium

  * Nave se incendia en NIVEL5. Aplica a temas: Original y Steampunk

 -- Diego Accorinti <diegoacco@gmail.com>  Tue, 06 Nov 2018 11:30:13 -0300

tierraluna (0.2.6) unstable; urgency=medium

  * fix música on off y salir con ESC

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 01 Nov 2018 11:53:41 -0300

tierraluna (0.2.5) unstable; urgency=medium

  * fix demo loop

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 02 Aug 2018 12:09:33 -0300

tierraluna (0.2.4) unstable; urgency=medium

  * sistema de temas.  Tema nuevo SteamPunk

 -- Diego Accorinti <diegoacco@gmail.com>  Tue, 06 Mar 2018 12:08:00 -0300

tierraluna (0.2.3) unstable; urgency=medium

  * Asteroides con comportamientos únicos. Estallar. Congelar. Aceleración en nave.

 -- Diego Accorinti <diegoacco@gmail.com>  Wed, 10 Jan 2018 17:05:52 -0300

tierraluna (0.2.2) unstable; urgency=medium

  * Partida desde la tierra, llegada a la luna. Fondos con paralaje. 

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 28 Dec 2017 15:53:23 -0300

tierraluna (0.2.0) unstable; urgency=medium

  * enemigos nuevos en cada nivel. Aparición de ARSAT-2

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 07 Dec 2017 16:18:43 -0300

tierraluna (0.1.9-1) unstable; urgency=medium

  * fix menor 

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 15 Sep 2016 11:18:29 -0300

tierraluna (0.1.8-1) unstable; urgency=medium

  * fix detener música al perder 

 -- Diego Accorinti <diegoacco@gmail.com>  Wed, 14 Sep 2016 14:43:35 -0300

tierraluna (0.1.7-1) unstable; urgency=medium

  * Jugar de nuevo al perder
  * FIX del launcher menu (error utf-8) final
  * cambio texto final (sin vida x exánime)

 -- Diego Accorinti <diegoacco@gmail.com>  Wed, 14 Sep 2016 13:10:18 -0300

tierraluna (0.1.6-1) unstable; urgency=medium

  * fix utf8-codec 2

 -- Diego Accorinti <diegoacco@gmail.com>  Wed, 14 Sep 2016 10:29:32 -0300

tierraluna (0.1.5-1) unstable; urgency=medium

  * fix utf8-codec

 -- Diego Accorinti <diegoacco@gmail.com>  Tue, 13 Sep 2016 16:35:17 -0300

tierraluna (0.1.4-1) unstable; urgency=medium

  * fix lanzador

 -- Diego Accorinti <diegoacco@gmail.com>  Tue, 13 Sep 2016 14:48:37 -0300

tierraluna (0.1.3-1) unstable; urgency=medium

  * fix del lanzador

 -- Diego Accorinti <diegoacco@gmail.com>  Tue, 13 Sep 2016 12:31:59 -0300

tierraluna (0.1.2-1) unstable; urgency=medium

  * tierraluna.desktop fix

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 12 Sep 2016 15:00:16 -0300

tierraluna (0.1.1-1) unstable; urgency=medium

  * RC 1

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 12 Sep 2016 10:47:00 -0300

tierraluna (0.1.0-3) unstable; urgency=medium

  * cambio menor 

 -- Diego Accorinti <diegoacco@gmail.com>  Fri, 09 Sep 2016 17:36:27 -0300

tierraluna (0.1.0-2) unstable; urgency=medium

  * huevo de pascua 

 -- Diego Accorinti <diegoacco@gmail.com>  Fri, 09 Sep 2016 17:06:59 -0300

tierraluna (0.1.0-1)  unstable; urgency=medium

  * release candidate 1 Huayra 4

 -- Diego Accorinti <diegoacco@gmail.com>  Fri, 09 Sep 2016 15:31:23 -0300

tierraluna (0.0.15-3) unstable; urgency=medium

  * prueba firmado 2 

 -- Diego Accorinti <diegoacco@gmail.com>  Fri, 09 Sep 2016 14:29:27 -0300

tierraluna (0.0.15-2) unstable; urgency=medium

  * prueba firmado

 -- Diego Accorinti <diegoacco@gmail.com>  Fri, 09 Sep 2016 13:23:18 -0300

tierraluna (0.0.15-1) unstable; urgency=medium

  * Cambios menores

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 08 Sep 2016 15:09:43 -0300

tierraluna (0.0.14-1) unstable; urgency=medium

  * Prueba Strider 2 

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 01 Sep 2016 16:38:07 -0300

tierraluna (0.0.13-2) unstable; urgency=medium

  * Prueba Strider

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 01 Sep 2016 16:33:08 -0300

tierraluna (0.0.13-1) unstable; urgency=medium

  * cambio menor

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 29 Aug 2016 11:06:59 -0300

tierraluna (0.0.12-2) unstable; urgency=medium

  * Prueba empaquetado 2

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 29 Aug 2016 10:59:15 -0300

tierraluna (0.0.12-1) unstable; urgency=medium

  * prueba empaquetado 1

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 29 Aug 2016 10:42:12 -0300

tierraluna (0.0.11-1) unstable; urgency=medium

  * cambios menores

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 29 Aug 2016 10:12:14 -0300

tierraluna (0.0.10-1) unstable; urgency=medium

  * Pantalla FINAL
  * 5 niveles distintos
 
 -- Diego Accorinti <diegoacco@gmail.com>  Fri, 26 Aug 2016 16:45:12 -0300

tierraluna (0.0.9-1) unstable; urgency=medium

  * Musica 
  * leyendas
  * textos externos

 -- Diego Accorinti <diegoacco@gmail.com>  Fri, 12 Aug 2016 16:57:27 -0300

tierraluna (0.0.8-1) unstable; urgency=medium

  * Non-maintainer upload.
 
 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 08 Aug 2016 16:11:14 -0300

tierraluna (0.0.7-1) unstable; urgency=medium

  * prueba

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 08 Aug 2016 16:00:07 -0300

tierraluna (0.0.5-1) unstable; urgency=medium

  * prueba empaquetado

 -- Diego Accorinti <diegoacco@gmail.com>Diego Accorinti <diegoacco@gmail.com>  Mon, 08 Aug 2016 12:15:22 -0300

tierraluna (0.0.4-2) unstable; urgency=medium

  * prueba 

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 08 Aug 2016 12:10:11 -0300

tierraluna (0.0.4-1) unstable; urgency=medium

  * se agregó manpage

 -- Diego Accorinti <diegoacco@gmail.com>  Mon, 08 Aug 2016 11:32:20 -0300

tierraluna (0.0.3-1) unstable; urgency=medium

  * release 2 

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 04 Aug 2016 15:31:24 -0300

tierraluna (0.0.2) UNSTABLE; urgency=medium

  * Versión inicial.

 -- Diego Accorinti <diegoacco@gmail.com>  Thu, 12 Jul 2016 17:09:18 +0100


juego.py
#!/usr/bin/env python
# -*- coding: utf-8
import pilasengine
from emisorHUMO import *
import os

# Inicializamos PILAS-ENGINE
# pasando como parámetros la resolución gráfica de la ventana y el título de la misma:

pilas = pilasengine.iniciar(ancho=900, alto=550, titulo='TierraLuna')


# Habilitando el Audio en PILAS: 

try:
  pilas.forzar_habilitacion_de_audio()
except AttributeError:
  print u"Omitiendo Habilitación forzada de audio, version anterior a 1.4.8".encode('utf-8')


# Definimos la ruta hasta los archivos de las dos tipografías que utilizaremos en el juego. 

ruta = os.path.dirname(os.path.realpath(__file__))
url_fuente = ruta + '/Tentacles.ttf'
url_fuente2 = ruta + '/Oswald-Regular.ttf'


# Declaramos Luna. Este actor es utilizado en la presentación.

class Luna(pilasengine.actores.Actor):
	
	def iniciar(self):
		url = ruta + '/imagenes/luna.jpg'
		self.imagen = url
		self.x = 0
		self.y = -700
		self.escala = 1.3
		self.z = -1

	def actualizar(self):
		self.rotacion -= 0.05


class Arsat(pilasengine.actores.Actor):
	
	def iniciar(self):
		url = ruta + '/imagenes/ARSAT-2.png'
		self.imagen = url

	def actualizar(self):
		self.rotacion -= 0.06
		self.x += 0.6
		# Elimina el objeto cuando sale de la pantalla.
		if self.x > 600:
			self.eliminar()
		

			
class HUDArsat(pilasengine.actores.Actor):
	
	def iniciar(self):
		url = ruta + '/imagenes/ARSAT-2-HUD.png'
		self.imagen = url
		self.x = -600
		self.y = pilas.azar(-50, 150)
		self.escala = 1
		self.z = -1

	def actualizar(self):
		self.x += 0.6
		# Elimina el objeto cuando sale de la pantalla.
		if self.x > 600:
			self.eliminar()
		


# Esta escena es el juego propiamente. Es lo que comenzará cuando elijamos "iniciar juego" en el menú principal

class PantallaJuego(pilasengine.escenas.Escena):

	velocidad_asteroides = 2


	class Asteroide(pilasengine.actores.Actor):

		def iniciar(self, tipo):
			self.tipo = tipo
			if self.tipo == "uno":
				self.imagen = ruta + '/imagenes/asteroide.png'
				self.giro = 1
			if self.tipo == "dos":
				self.imagen = ruta + '/imagenes/asteroide2.png'
				self.giro = 3
			if self.tipo == "tres":
				self.imagen = ruta + '/imagenes/asteroide3.png'
				self.giro = 2
			if self.tipo == "cuatro":
				self.imagen = ruta + '/imagenes/asteroide4.png'
				self.giro = -2
			if self.tipo == "cinco":
				self.imagen = ruta + '/imagenes/asteroide5.png'
				self.giro = 3


			self.escala = 0.3
			self.x = -500
			self.y = pilas.azar(-300, 300)
			self.z = self.y

		def actualizar(self):
			self.rotacion += self.giro
			self.x += PantallaJuego.velocidad_asteroides
			# Elimina el objeto cuando sale de la pantalla.
			if self.x > 500:
				self.eliminar()

	#defino un grupo de enemigos
	enemigos = pilas.actores.Grupo()

	def crear_asteroide_uno(self):
		#creo el actor enemigo
		asteroide = self.Asteroide(pilas, tipo="uno");
		#creo un objeto para la física
		c1 = pilas.fisica.Circulo(asteroide.x, asteroide.y, 150, restitucion=1, amortiguacion=2)
		asteroide.imitar(c1)
		#lo agrego al grupo
		self.enemigos.agregar(asteroide)
		
	def crear_asteroide_dos(self):
		asteroide = self.Asteroide(pilas, tipo="dos");
		c1 = pilas.fisica.Circulo(asteroide.x, asteroide.y, 150, restitucion=1, amortiguacion=1)
		asteroide.imitar(c1)
		self.enemigos.agregar(asteroide)

	def crear_asteroide_tres(self):
		asteroide = self.Asteroide(pilas, tipo="tres");
		c1 = pilas.fisica.Circulo(asteroide.x, asteroide.y, 150, restitucion=1, amortiguacion=3)
		asteroide.imitar(c1)
		self.enemigos.agregar(asteroide)
		
	def crear_asteroide_cuatro(self):
		asteroide = self.Asteroide(pilas, tipo="cuatro");
		c1 = pilas.fisica.Circulo(asteroide.x, asteroide.y, 150, restitucion=1, amortiguacion=2)
		asteroide.imitar(c1)
		self.enemigos.agregar(asteroide)
		
	def crear_asteroide_cinco(self):
		asteroide = self.Asteroide(pilas, tipo="cinco");
		c1 = pilas.fisica.Circulo(asteroide.x, asteroide.y, 150, restitucion=1, amortiguacion=2)
		asteroide.imitar(c1)
		self.enemigos.agregar(asteroide)
		
	def iniciar(self):

		global contador_texto
		global contador_choques
		global flag
		global flagEspeciales
		flag = [False, False, False, False, False]
		flagEspeciales = [False]


		contador_texto = 0
		contador_choques = 0

		fondo = pilas.fondos.Galaxia(dx=-2, dy=0)
		# MUSICA
		url = ruta + '/data/Dreams-Become-Real.ogg'
		musica = pilas.sonidos.cargar(url)
		musica.reproducir(repetir=True)

		# Boton de sonido ON / OFF
		boton_musica = pilas.actores.Boton();
		url = ruta + '/imagenes/sonidoON.png'
		boton_musica.imagen = url
		boton_musica.x = 410
		boton_musica.y = 240
		boton_musica.z = -1000
		boton_musica.sonidoOnOff = True

		def cambio(sonidoOnOff):

			if boton_musica.sonidoOnOff:
				# apago la musica
				musica.detener()
				url = ruta + '/imagenes/sonidoOFF.png'
				boton_musica.imagen = url
				boton_musica.sonidoOnOff = False

			else:
				#enciendo la música
				musica.reproducir(repetir=True)
				url = ruta + '/imagenes/sonidoON.png'
				boton_musica.imagen = url
				boton_musica.sonidoOnOff = True

		boton_musica.conectar_presionado(cambio, boton_musica.sonidoOnOff)

		# LOS TEXTOS
		#Le pido la biblioteca de textos contenido en textos.py
		from textos import textos

		texto_personalizado = pilas.actores.Texto('', magnitud=31, fuente= url_fuente, y= -230, ancho = 230)
		sombra_texto_personalizado = pilas.actores.Texto('', magnitud=31, fuente= url_fuente, y= -233, x=1, ancho = 230)

		sombra_texto_personalizado.color = pilas.colores.negro
		sombra_texto_personalizado.z = 4

		class Nave(pilasengine.actores.Actor):

			def iniciar(self):
				url = ruta + '/imagenes/lanave.png'
				self.imagen = url

		class Astronauta(pilasengine.actores.Actor):

			def iniciar(self):
				url = ruta + '/imagenes/astronauta.png'
				self.imagen = url

			def actualizar(self):
				self.rotacion += 1

		minave = Nave(pilas);
		minave.z = -2


		c2 = pilas.fisica.Circulo(minave.x, minave.y, 70, restitucion=0.1, amortiguacion=0.5)
		def seguir(evento):
			#print "X: " + str(evento.x)
			#print "Y: " + str(evento.y)
			if (evento.x < 390) or (evento.y < 219):
				# Solo sigo si el click es fuera del icono de sonido
				empujarx = (evento.x - c2.x) / 8
				empujary = (evento.y - c2.y) / 8
				c2.empujar(empujarx,empujary)

		def frenar():
			c2.velocidad_x /= 2
			c2.velocidad_y /= 2

		self.frenar = frenar
		pilas.tareas.siempre(0.1, self.frenar)


		self.seguir = seguir
		pilas.eventos.click_de_mouse.conectar(self.seguir)

		minave.imitar(c2, con_rotacion=False)

		emisor = EmisorHUMO(pilas, 0, 0)
		url = ruta + '/imagenes/humo.png'
		emisor.imagen_particula = pilas.imagenes.cargar_grilla(url)
		emisor.constante = True
		emisor.composicion = "blanco"
		emisor.duracion = 2
		emisor.frecuencia_creacion = 0.05
		emisor.vida = 3
		emisor.aceleracion_x_min = 36
		emisor.aceleracion_x_max = 50
		emisor.x_min = 171
		emisor.y_min = 2
		emisor.transparencia_min = 30
		emisor.transparencia_max = 50


		emisor.aprender(pilas.habilidades.Imitar, minave)

		minave.aprender(pilas.habilidades.MoverseConElTeclado)
		minave.aprender(pilas.habilidades.LimitadoABordesDePantalla)


		def imprimir_texto():

			global contador_texto

			#cambio el texto
			texto_personalizado.ancho = 900
			sombra_texto_personalizado.ancho = 900
			texto_personalizado.texto = textos[contador_texto]
			sombra_texto_personalizado.texto = textos[contador_texto]
			#oculto el texto y su sombra
			texto_personalizado.transparencia = 100
			sombra_texto_personalizado.transparencia = 100
			#lo hago visible nuevamente
			texto_personalizado.transparencia = [0]
			sombra_texto_personalizado.transparencia = [0]

			# Centro los textos en la pantalla

			factor = len(textos[contador_texto]) * 7

			texto_personalizado.x = 450 - factor
			sombra_texto_personalizado.x = 450 - factor

			contador_texto += 1 # incremento el contador para que la próxima vez muestre el siguiente texto.


		# Creo una tarea para que aparezcan los textos, cada 5 segundos.
		tareaMostrarTextos = pilas.tareas.siempre(5, imprimir_texto)


		def nave_choco():



			global contador_choques

			pilas.camara.vibrar(3, 0.5)

			contador_choques += 1

			if contador_choques == 2:
				minave.imagen = ruta + '/imagenes/lanave_01.png'
				emisor.frecuencia_creacion = 0.07
			if contador_choques == 4:
				minave.imagen = ruta + '/imagenes/lanave_02.png'
				emisor.frecuencia_creacion = 0.10
			if contador_choques == 7:
				minave.imagen = ruta + '/imagenes/lanave_03.png'
				emisor.frecuencia_creacion = 0.13
			if contador_choques == 9:
				minave.imagen = ruta + '/imagenes/lanave_04.png'
				emisor.eliminar()
				minave.rotacion = [360], 2
			if contador_choques == 12:
				pilas.camara.x = minave.x
				pilas.camara.y = minave.y
				perdido = Astronauta(pilas);
				minave.eliminar()
				musica.detener()
				pilas.camara.escala = [1.2, 1.5, 1]
				perdido.escala = [1, 0.4]
				texto_personalizado3 = pilas.actores.Texto(u'por miles de años flotarás exánime en el espacio · presiona ESPACIO', magnitud=18, fuente= url_fuente2, y= -230, x = 0)
				texto_personalizado3.color =  pilas.colores.blanco
				tareaMostrarTextos.terminar()
				texto_personalizado.transparencia = 100
				sombra_texto_personalizado.transparencia = 100
				pilas.eventos.pulsa_tecla.conectar(self.al_pulsar_tecla)



			mensajeNeo = [False, False]
			print "Choques = " + str(contador_choques)
			if (contador_choques == 8) and (mensajeNeo[0] == False):
				os.system('clear')
				print "Despierta, Neo."
				print "La Matrix te tiene."
				print "Sigue al conejo blanco."
				print "Toc toc, Neo."
				mensajeNeo[0] = True
			if (contador_choques == 9) and (mensajeNeo[1] == False):
				os.system('clear')
				print "Choques = " + str(contador_choques)
				mensajeNeo[1] = True


		# Creo un control de coliciones para saber cuando perdes
		pilas.colisiones.agregar(minave, self.enemigos, nave_choco)

		#Elimino los límites laterales y la gravedad
		pilas.fisica.gravedad_x = 0
		pilas.fisica.gravedad_y = 0
		pilas.fisica.eliminar_paredes()
		pilas.fisica.eliminar_techo()
		pilas.fisica.eliminar_suelo()

	# Cuando pierdo, si presiono una tecla termina el juego y se cierra
	def al_pulsar_tecla(self, tecla):
		global flag
		#print tecla.codigo
		#pilas.escenas.PantallaMenu()
		if tecla.codigo == 32:
			flag = [False, False, False, False, False]
			self.pilas.escenas.PantallaMenu()




	def actualizar(self):
		''' Acá definimos las distintas etapas del juego, según van avanzando los textos
		    podemos ir cambiando los enemigos, el fondo, etc.  '''
		global contador_texto
		global flag

		def cambio_nivel(nivel, leyenda):# Cuando pasamos de nivel

			if (nivel <> 1): pilas.camara.vibrar(4, 1)
			texto_nivel = pilas.actores.Texto(cadena_de_texto="Nivel " + str(nivel) + ": " + leyenda,
			 magnitud = 40, x = -400, y = 230)
			texto_nivel.centro = ("izquierda", "centro")
			texto_nivel.transparencia = 0
			texto_nivel.transparencia = [100],15
			texto_nivel.escala = [0.7],10
			texto_nivel.x = [-400,-430],10
			texto_nivel.y = [240],10


		if contador_texto == 1:
			''' Recien al segundo texto comienzan a venir los asteroides '''
			# Creo una tarea para que aparezca un asteroide cada 2 segundos.

		if (flag[0]) == False:
				print "NIVEL 1"
				PantallaJuego.tareaAsteroides = pilas.tareas.siempre(2, self.crear_asteroide_uno)
				flag[0] = True
				cambio_nivel(1, "Cinco puntos de luz")

		if contador_texto == 34:
			''' ###  NIVEL 2 ###
			Llegamos al segundo nivel.  Aumentamos la velocidad de los enemigos y cambiamos el fondo. '''
			# Creo una tarea para que aparezca un asteroide cada 1.5 segundos.
			if (flag[1]) == False:
				print "NIVEL 2"
				PantallaJuego.tareaAsteroides.terminar()
				PantallaJuego.tareaAsteroides = pilas.tareas.siempre(1.5, self.crear_asteroide_dos)
				PantallaJuego.velocidad_asteroides = 6
				fondo = pilas.fondos.Galaxia(dx=-3, dy=0)
				fondo.imagen = ruta + '/imagenes/galaxia_02.png'
				cambio_nivel(2, "Constelaciones")
				flag[1] = True

		if contador_texto == 68:
			''' ###  NIVEL 3 ### '''
			if (flag[2]) == False:
				print "NIVEL 3"
				PantallaJuego.tareaAsteroides.terminar()
				PantallaJuego.tareaAsteroides = pilas.tareas.siempre(1.5, self.crear_asteroide_tres)
				PantallaJuego.velocidad_asteroides = 4
				fondo = pilas.fondos.Galaxia(dx=-2, dy=0)
				fondo.imagen = ruta + '/imagenes/galaxia_03.png'
				cambio_nivel(3, "Nuestra casa")
				flag[2] = True
		if contador_texto == 102:
			''' ###  NIVEL 4 ### '''
			if (flag[3]) == False:
				print "NIVEL 4"
				PantallaJuego.tareaAsteroides.terminar()
				PantallaJuego.tareaAsteroides = pilas.tareas.siempre(2, self.crear_asteroide_cuatro)
				PantallaJuego.velocidad_asteroides = 2
				fondo = pilas.fondos.Galaxia(dx=-1, dy=0)
				fondo.imagen = ruta + '/imagenes/galaxia_04.png'
				cambio_nivel(4, "Mirando al pasado")
				flag[3] = True
		if contador_texto == 130:
			''' ###  NIVEL 5 ### '''
			if (flag[4]) == False:
				print "NIVEL 5"
				PantallaJuego.tareaAsteroides.terminar()
				PantallaJuego.tareaAsteroides = pilas.tareas.siempre(1.5, self.crear_asteroide_cinco)
				PantallaJuego.velocidad_asteroides = 6
				fondo = pilas.fondos.Galaxia(dx=-3, dy=0)
				fondo.imagen = ruta + '/imagenes/galaxia_05.png'
				cambio_nivel(5, "La llegada")
				flag[4] = True
		if contador_texto == 141:
			''' FINAL! Ganó el juego '''
			pilas.escenas.PantallaFinal()



		''' Eventos únicos especiales durante el juego '''	

		# paso de ARSAT-2
		if contador_texto == 16:
			if flagEspeciales[0] == False:
				
				hudarsat = HUDArsat(pilas)
				hudarsat.z = 1000
				arsat = Arsat(pilas)
				arsat.x = hudarsat.x
				arsat.y = hudarsat.y
				arsat.z = 999
				flagEspeciales[0] = True
			



			

class PantallaFinal(pilasengine.escenas.Escena):
	def iniciar(self):
		fondo = pilas.fondos.Fondo()
		url = ruta + '/imagenes/final.jpg'
		fondo.imagen = pilas.imagenes.cargar(url)
		texto_personalizado = pilas.actores.Texto(u'¡ganaste!', magnitud=60, fuente= url_fuente,
		 y= -50, x = 20)



class PantallaConfig(pilasengine.escenas.Escena):

	mostre_huevo_pascua = False

	def iniciar(self):

		fondo = pilas.fondos.Galaxia(dx=0, dy=0)
		fondo.imagen = ruta + '/imagenes/fondo-config.png'
		texto_personalizado = pilas.actores.Texto(u'¡no hay nada que configurar!', magnitud=55, fuente= url_fuente,
		 y= 0, x = 0)
		texto_personalizado2 = pilas.actores.Texto(u'presione ESPACIO para continuar', magnitud=14, fuente= url_fuente2, y= -230, x = 0)
		texto_personalizado2.color =  pilas.colores.gris



		pilas.eventos.pulsa_tecla.conectar(self.al_pulsar_tecla)

	def al_pulsar_tecla(self, tecla):

		if tecla.codigo == 32:
			pilas.escenas.PantallaMenu()
		else:
			if (self.mostre_huevo_pascua == False):
				texto_personalizado3 = pilas.actores.Texto(u'Un juego de Diego Accorinti para Huayra gnu/linux', magnitud=12, fuente= url_fuente2, y= -200, x = 0)
				self.mostre_huevo_pascua = True

# Escena Menu
def cargar_escena_juego():
	pilas.escenas.PantallaJuego()

def cargar_escena_config():
	pilas.escenas.PantallaConfig()

def salir_del_juego():
	pilas.terminar()


class PantallaMenu(pilasengine.escenas.Escena):

	def iniciar(self):
		fondo = pilas.fondos.Color(pilas.colores.negro)
		fondo = pilas.fondos.Fondo()
		url = ruta + '/imagenes/intro.png'
		fondo.imagen = pilas.imagenes.cargar(url)
		fondo.z = -2

		luna = Luna(pilas);

		menu = pilas.actores.Menu([
					('JUGAR', cargar_escena_juego),
					(u'CONFIG', cargar_escena_config),
					('SALIR', salir_del_juego),
				], fuente = url_fuente2, y = 150)
		menu.escala = 1
		menu.x = [300],3
		menu.transparencia = 100
		menu.transparencia = [0],3


pilas.escenas.vincular(PantallaJuego)
pilas.escenas.vincular(PantallaMenu)
pilas.escenas.vincular(PantallaConfig)
pilas.escenas.vincular(PantallaFinal)
pilas.escenas.PantallaMenu()

pilas.ejecutar()
